src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -973,16 +970,35 @@ public function getDirty()
     }
 
     /**
-     * Determine if the new and old values for a given key are numerically equivalent.
+     * Determine if the new and old values for a given key are equivalent.
      *
-     * @param  string  $key
+     * @param  string $key
+     * @param  mixed  $current
      * @return bool
      */
-    protected function originalIsNumericallyEquivalent($key)
+    protected function originalIsEquivalent($key, $current)
     {
-        $current = $this->attributes[$key];
+        if (! array_key_exists($key, $this->original)) {
+            return false;
+        }
 
-        $original = $this->original[$key];
+        if ($current === $original = $this->getOriginal($key)) {
+            return true;
+        }
+
+        // When check rich this check and current attribute value not equals with original, we should skip next steps
+        // if current is null
+        if (is_null($current)) {
```

Comment:
[Fully agree](https://github.com/octobercms/library/commit/565761d876447b8f2819a0fa1ab48354599eb814#diff-dfa33b774dcd9386d95be910586cc883R354), either that or should to include the original value in the check too:

```
 if (is_null($current) || is_null($original)) {
```

- `is_null($current)` covers updating filled column with NULL
- `is_null($original)` covers updating NULL with filling